### PR TITLE
User

### PR DESCRIPTION
**Pull request by ManasviniArul**
user credentials for banking system

### CODE REVIEW

**Review by grewaviv — CHANGES_REQUESTED** [review]
Please refer the comment in https://github.com/khoros-campus-hires/BankingSystem/pull/2

**Comment by grewaviv** [comment]
Please refer https://github.com/khoros-campus-hires/BankingSystem/pull/2/

**Comment by ManasviniArul** [comment]
> Please refer #2

okay will check

File: Banking/src/main/java/com/example/Banking/Exception/ErrorHandler.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.example.Banking.Exception;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+
+public class ErrorHandler {
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/Exception/ErrorHandler.java]: I thought Error handling was already done as part of dependent PR?

File: Banking/src/main/java/com/example/Banking/Exception/ErrorHandler.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.example.Banking.Exception;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/Exception/ErrorHandler.java]: Check previous comments there shouldn't be space between annotation and class

File: Banking/src/main/java/com/example/Banking/service/UserDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+import java.util.List;
+
+@Component
+public class UserDaoImpl implements UserDao {
+
+    @Autowired
+    UserRepository userRepository;
+
+    private static final String ERRORMESSAGE = "user not found for this id";
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/UserDaoImpl.java]: `ERROR_MESSAGE`

File: Banking/src/main/java/com/example/Banking/Exception/IdNotFound.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Data;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@Data
+@ResponseStatus(value = HttpStatus.NOT_FOUND)
+@Component
+
+public class IdNotFound extends Exception {
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/Exception/IdNotFound.java]: Is it used anywhere?

File: Banking/src/main/java/com/example/Banking/service/UserDaoImpl.java (hunk 3 above)
Comment (ManasviniArul) [reply Banking/src/main/java/com/example/Banking/service/UserDaoImpl.java]: changed

File: Banking/src/main/java/com/example/Banking/Exception/IdNotFound.java (hunk 4 above)
Comment (ManasviniArul) [reply Banking/src/main/java/com/example/Banking/Exception/IdNotFound.java]: this is also part of exception. by mistake committed exception files. removed them now.

File: Banking/src/main/java/com/example/Banking/model/User.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+@AllArgsConstructor
+public class User {
+
+        @Id
+        private long id;
+        private String firstName;
+        private String lastName;
+        private Date dob;
+        private String address;
+        private String contact_number;
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/model/User.java]: `contactNumber` camel casing

File: Banking/src/main/java/com/example/Banking/service/UserService.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    public void createUser(User user) {
+        userDao.insertUser(user);
+    }
+
+    public void deleteUser(Long Id) throws IdNotFoundException {
+        userDao.deleteUser(Id);
+    }
+
+    public void updateUser(User user) throws Exception{
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/UserService.java]: `Exception {`

**Review by grewaviv — APPROVED** [review]
You can merge after the minor corrections i suggested

**Pull request merged**